tests/test_tables.py

```diff
@@ -214,3 +214,40 @@ def test_add_lines():
     tab2 = page.find_tables(add_lines=more_lines)[0]
     assert tab2.col_count == 4
     assert tab2.row_count == 5
+
+
+def test_3148():
+    """Ensure correct extraction text of rotated text."""
+    doc = fitz.open()
+    page = doc.new_page()
+    rect = fitz.Rect(100, 100, 300, 300)
+    text = (
+        "rotation 0 degrees",
+        "rotation 90 degrees",
+        "rotation 180 degrees",
+        "rotation 270 degrees",
+    )
+    degrees = (0, 90, 180, 270)
+    delta = (2, 2, -2, -2)
+    cells = fitz.make_table(rect, cols=3, rows=4)
+    for i in range(3):
+        for j in range(4):
+            page.draw_rect(cells[j][i])
+            k = (i + j) % 4
+            page.insert_textbox(cells[j][i] + delta, text[k], rotate=degrees[k])
+    # doc.save("multi-degree.pdf")
+    tabs = page.find_tables()
+    tab = tabs[0]
+    for extract in tab.extract():
+        for item in extract:
+            item = item.replace("\n", " ")
+            assert item in text
```

Comment:
Might be good to check that we find each of the items in `text`. E.g.:
```
text_set = set(text)
...
text_set.remove(item)
```

[It looks like `tab.extract()` just returns the text; could it also return the rotation (0, 90, 180, 270)? Then we could additionally check that the rotation matches the text.]

Reply:
No, the format of `table.extract()` is fixed for the time being.

The test that the extracted text of each cell fully matches the multi-word original content already proves that I got everything. If anything would go wrong with a non-0 rotation, the character and / or word sequence would be spoiled - and I am excluding this with that test: IAW the fact that text "rotation by 270 degrees" is extracted literally like this, already is **sufficient proof** that extraction of this angle did work.

If we want to be overly pedantic, we might use a running index and in addition assert the right item in the original array `text`- which has been cyclically reused to originally fill the table.

> I purposefully chose a column count that is not a multiple of 4 to ensure that each rotation occurs in every column at least once.

But that would test a different thing - outside the test scope here, namely that all text items have been extracted in the same sequence as originally inserted (as opposed to only testing that we don't have problems with any of the 4 angles). This

My goal is to ultimately only use our own text extraction and especially text / word re-composition. That would allow for fully reliably extracting all text meta-information (color, font, font size, etc.)